hunts/models.py

```diff
@@ -22,6 +26,15 @@ def save(self, *args, **kwargs):
             self.slug = slugify(self.name)
         super().save(*args, **kwargs)
 
+    def clean(self):
+        # adds default start and end times if not provided
+        new_start_time = self.start_time
```

Comment:
Why not just:

```
if not self.start_time:  
    self.start_time = self.created_on  
if not self.end_time:
    self.end_time = self.start_time + timedelta(days=3)
```

Reply:
The code yells at me because the fields might be empty, so start_time + timedelta might be undefined. This way, all the changes are committed at once. It's kinda hacky, but I couldn't find a cleaner way to do it.

Reply:
I'm confused -- the code snippet in my previous comment should be functionally the same as what you've written (sans a temp variable which I *thought* was unnecessary). Am I missing something?

Even if `start_time` is undefined, it should be guaranteed to be defined by the last line.

Reply:
Weirdly, `start_time` didn't seem to be defined until `save` was called (after `clean` finished). One fix I was thinking of was calling `save` again between confirming `start_time` and `end_time`, but that seems messy.

Reply:
Here's a fix. It's still a bit messy, but it's cleaner than before. `clean` doesn't have to be called, and `save` only needs to be called once.

```
    def save(self, *args, **kwargs):
        if not self.slug:
            self.slug = slugify(self.name)
        super().save(*args, **kwargs)

        # adds default start and end times if not provided
        if not self.start_time:
            self.start_time = self.created_on
            super().save(*args, **kwargs)
        if not self.end_time:
            self.end_time = self.start_time + timedelta(days=3)
            super().save(*args, **kwargs)
```